### PR TITLE
Add comprehensive unit tests for PIRSensorHelper

### LINKED ISSUE #309: Write Unit Tests for PIRSensorHelper
**Description:**
The PIRSensorHelper class manages a PIR motion sensor using Pi4J's DigitalInput. Unit tests are required to validate its behavior, including motion detection, event listener management, and initialization.

**Developer Notes:**

1. Test Initialization Logic:
  
  Verify that isMoving is correctly initialized based on pirSensorInput.isHigh().
  Ensure that the initialize method adds an event listener to pirSensorInput.

2. Test Motion Detection (initialize Method):
  
  Simulate a change in pirSensorInput state and check if isMoving updates correctly.

3. Test Event Listener Management:

  Validate that addEventListener correctly adds an event listener to pirSensorInput.
  Ensure removeEventListener removes the listener and sets pirSensorInputListener to null.
  Check that multiple calls to removeEventListener do not cause errors.

4. Mocking Dependencies:

  Use Mockito to mock DigitalInput and DigitalStateChangeListener.
  Ensure tests do not require real hardware.

5. Edge Cases:
  
  Verify behavior when pirSensorInput is initially low.
  Ensure addEventListener correctly overrides existing listeners.
  Test that removeEventListener does not fail when no listener is present.

**Acceptance Criteria**
- [ ] All test cases outlined in the tasks above are implemented.

- [ ] Code coverage for the MicroSwitchHelper class is at least 70%.

- [ ] Tests are written using a Java JUnit testing framework

- [ ] Mocking is implemented using a framework such as Mockito or a similar tool.

- [ ] All tests pass successfully, and edge cases are handled gracefully.

### PR DESCRIPTION
**Pull request by Ed0827**
Fixes #309 

What was changed:
1. Added unit tests for the PIRSensorHelper in PIRSensorHelperTest.
2. Tests cover initialization logic, motion detection, and event listener management.
3. Edge cases such as an initially low sensor state, overriding event listeners, and multiple calls to removeEventListener have been verified.

Why was it changed:
1. To improve the reliability and maintainability of the PIRSensorHelper functionality.
2. Enhanced test coverage ensures that the sensor behavior is validated without needing actual hardware.
3. The changes help prevent regressions and simplify future development.

How was it changed:
1. Implemented tests using JUnit and Mockito to mock dependencies like DigitalInput and DigitalStateChangeListener.
2. The tests simulate sensor state changes to validate that 'isMoving' updates correctly.
3. Added tests to ensure that addEventListener properly overrides existing listeners and that multiple calls to removeEventListener are handled gracefully.

### CODE REVIEW

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The unit tests in PIRSensorHelperTest are well-structured, but the assertion messages can be improved for better clarity. For instance, instead of "Expected isMoving to be true when DigitalInput is high", use "Expected isMoving to be true when PIR sensor detects motion (DigitalInput is high)". This makes the test output more informative. Please update the assertion messages accordingly and push the changes. The rest of the implementation looks great! 👍

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.mockito.ArgumentCaptor;
+
+class PIRSensorHelperTest {
+    private DigitalInput pirSensorInput;
+    private PIRSensorHelper pirSensorHelper;
+
+    @BeforeEach
+    void setUp() {
+        pirSensorInput = mock(DigitalInput.class);
+        when(pirSensorInput.isHigh()).thenReturn(true);
```
Comment (avaenk) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java:23]: Be sure this line does not cause any undefined behavior in your other tests, I tried a similar thing in my PR and noticed some strange behavior in my tests and ended up removing it. If it doesn't cause any issues feel free to keep.

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        pirSensorHelper = new PIRSensorHelper(pirSensorInput);
+    }
+
+    @Test
```
Comment (avaenk) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java:27]: Maybe you could create one function for the test is moving instead of re-declaring the pirSensorHelper multiple times. It's possible to find a way to combine these functions into one, have the same functionality, and only have a few more lines of code.

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        assertFalse(pirSensorHelper.isMoving, "Expected isMoving to update to false after state change.");
+
+        when(pirSensorInput.isHigh()).thenReturn(true);
+        internalListener.onDigitalStateChange(null);
+        assertTrue(pirSensorHelper.isMoving, "Expected isMoving to update to true after state change.");
+    }
+
+    @Test
+    void testMultipleRemoveEventListenerCalls() {
+        pirSensorHelper.removeEventListener();
```
Comment (avaenk) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java:93]: For our testing purposes I see why you use only two calls to the event listener, however, on your own time before re-committing the code ensure that 4-5 calls to the event listener causes no undefined behavior, two calls may not be enough to be sure.

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    void testIsMovingFieldSetCorrectlyWhenInputIsLow() {
+        when(pirSensorInput.isHigh()).thenReturn(false);
+        pirSensorHelper = new PIRSensorHelper(pirSensorInput);
+        assertFalse(pirSensorHelper.isMoving, "Expected isMoving to be false when DigitalInput is low");
+    }
+
+    @Test
+    void testInitializationLogsCaptureSystemOut() {
+        PrintStream originalOut = System.out;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
Comment (avaenk) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PIRSensorHelperTest.java:44]: Love the use of ByteArrayOutputStream, awesome!

**Review by avaenk — COMMENTED** [review]
Great job!

**Review by yrlmanoharreddy — APPROVED** [review]
Great job on the unit tests for PIRSensorHelper! ✅ The tests are well-structured, comprehensive, and ensure the expected behavior. Everything looks good—approved! 🚀

**Comment by Ed0827** [comment]
> Great job on the unit tests for PIRSensorHelper! ✅ The tests are well-structured, comprehensive, and ensure the expected behavior. Everything looks good—approved! 🚀

Awesome!
